internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -52,10 +45,11 @@ func New(sess *session.Session) CloudFormation {
 // If the change set to create the stack cannot be executed, returns a ErrNotExecutableChangeSet.
 // Otherwise, returns a wrapped error.
 func (cf CloudFormation) DeployEnvironment(env *archer.DeployEnvironmentInput) error {
-	return cf.deploy(newEnvStackConfig(env, cf.box))
+	return cf.Deploy(newEnvStackConfig(env, cf.box))
 }
 
-func (cf CloudFormation) deploy(stackConfig stackConfiguration) error {
+// Deploy delpoys an entity that can be serialized into a Cloudformation template
+func (cf CloudFormation) Deploy(stackConfig archer.StackConfiguration) error {
```

Comment:
Let me know how much you hate this idea 💩

Instead of exporting this function,  should we create a separate`pipeline.go` file in this package just like we've done with `env.go`?  

The `cloudformation` package can take a dependency on the `manifest` package and create a separate file for each type that's exported from the `manifest` package.  So for example, I'm envisioning a 
`lbFargateStackConfig` struct that implements the `stackConfiguration` interface. For you, it could be a `pipelineStackConfig` struct.

If this doesn't make any sense let's sync up offline :)

Reply:
This is what I intentionally want to avoid because:
1. Use two data models to represent the pipeline. One only contains partial information while the other contains the full information (since we need to serialize it to CFN).
2. I want to keep the "main" function (in this case, the pipeline commands we are going to create) as simple as possible because it's harder to test.
3. I think it's very nice to be able to hide DB access (i.e. SSM) behind the data model. Similar to DynamoDBMapper.